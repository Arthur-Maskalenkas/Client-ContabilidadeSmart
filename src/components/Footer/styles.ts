import styled, { css } from 'styled-components'

import * as HeadingsStyles from 'components/Heading/styles'
import media from 'styled-media-query'
import theme from 'styles/theme'

export const Wrapper = styled.footer`
  ${HeadingsStyles.Wrapper} {
    text-transform: uppercase;
  }
`

export const Content = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(2, 1fr); // 1fr 1fr
    gap: ${theme.grid.gutter};
    margin-top: ${theme.spacings.medium};

    ${media.greaterThan('medium')`
      grid-template-columns: repeat(4, 1fr) // 1fr 1fr 1fr 1fr
    `}
  `}
`

export const Column = styled.div`
  ${({ theme }) => css`
    a,
    span {
      display: block;
      color: ${theme.colors.gray};
      text-decoration: none;
      margin-bottom: ${theme.spacings.xxsmall};
      font-size: ${theme.font.sizes.small};
    }
    a:hover {
      text-decoration: underline;
    }
  `}
`

export const Copyright = styled.div`
  ${({ theme }) => css`
    color: ${theme.colors.gray};
    font-size: ${theme.font.sizes.xsmall};
    margin-top: ${theme.spacings.large};
    margin-bottom: ${theme.spacings.medium};
    text-align: center;
  `}
`

export const Heading = styled(HeadingsStyles.Wrapper)`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.small};
  `}
`
