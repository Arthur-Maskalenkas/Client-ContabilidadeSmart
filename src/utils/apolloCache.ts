import { InMemoryCache } from '@apollo/client'
import { concatPagination } from '@apollo/client/utilities'

// A query QueryGames retorna -> games <-, e é isso que o cache vai acumular games

// ** fluxo:
// > Eu pego uma lista de free (retorna games assim como varias outras query)
// > o cache fica [game1,game2,game3]
// > Assim que outra parte usa alguma query que retorne game, ela vai receber independente de tudo [game1,game2,game3], e não [game4,game5,game6]
// **

// Para quebrar o fluxo, eu separei o cacheamento por 'where' e 'sort'. Assim fica algo como: (caso puxe 3 games de 3 queries diferentes)

//**
// > games: {"where: {"release_date_lte":"2021-01-29"}, "sort":"release_date:desc"}: [] 8 items"}
// > games: {"where: {"release_date_gt":"2021-01-29"}, "sort":"release_date:asc"}: [] 8 items"}
// > games: {"where: {"price": 0}}: [] 8 items
// > Antes ficaria tudo em games com 24 items
//**
export default new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        posts: concatPagination(['where', 'sort'])
      }
    }
  }
})
