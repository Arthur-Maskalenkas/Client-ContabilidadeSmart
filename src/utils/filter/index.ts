import { ParsedUrlQueryInput } from 'querystring'

type ParseArgs = {
  queryString: ParsedUrlQueryInput
}

export const parseQueryStringToWhere = ({ queryString }: ParseArgs) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const obj: any = {}

  // queryString = {platforms: ['windows','linux']}
  // filterItems = {name: 'platforms', type: 'checkbox'}
  Object.keys(queryString).forEach((key) => {
    // Pegando o valor e colocando dentro do objeto
    obj[key] = { slug_contains: queryString[key] }
  })

  return obj
}
