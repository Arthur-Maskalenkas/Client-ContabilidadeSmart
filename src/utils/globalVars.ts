export const IMAGE_NOT_FOUND =
  'https://res.cloudinary.com/dezwlfeyb/image/upload/v1635853776/102-022_800x_ebxk1t.jpg'

export const LINK_SITE = 'localhost:3000'

// --  API

export const API_LINKEDIN =
  'https://www.linkedin.com/shareArticle?mini=true&url='

export const API_FACEBOOK = 'https://www.facebook.com/sharer/sharer.php?u='

export const API_WHATSAPP = 'https://api.whatsapp.com/send?text='

// --  API
