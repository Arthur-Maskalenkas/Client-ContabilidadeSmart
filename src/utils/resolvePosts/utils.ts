export const operations = {
  nextPage: nextPageOperation(),
  backPage: backPageOperation()
}

type variablesType = {
  limit?: number
  start?: number
}

export type FetchMoreProps = {
  variables: variablesType
}
