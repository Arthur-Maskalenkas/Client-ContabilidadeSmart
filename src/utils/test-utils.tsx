import { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'

import { ThemeProvider } from 'styled-components'
import theme from 'styles/theme'

type CustomRenderProps = Omit<RenderOptions, 'queries'>

const customRender = (ui: ReactElement, { ...renderOptions }: CustomRenderProps = {}) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions)

// Tudo que eu chamar, vai vim dessa pasta aqui
export * from '@testing-library/react'
export { customRender as render }
