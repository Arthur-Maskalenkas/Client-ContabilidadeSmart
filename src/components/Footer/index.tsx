import Link from 'next/link'

import Heading from 'components/Heading'
import * as S from './styles'

const Footer = () => (
  <S.Wrapper>
    <S.Content>
      <S.Column aria-labelledby="column-contact">
        <S.Heading color="black" size="small">
          Contact Us
        </S.Heading>

        <a href="mailto:sac@wongames.com">mm@mm.com</a>
      </S.Column>

      <S.Column aria-labelledby="social-media">
        <S.Heading color="black">Follow us</S.Heading>

        <nav>
          <a
            href="https://www.instagram.com/mixeria-milionaria"
            target="_blank"
            rel="noopenner, noreferrer"
          >
            Instagram
          </a>
          <a
            href="https://www.twitter.com/mixeria-milionaria"
            target="_blank"
            rel="noopenner, noreferrer"
          >
            Twitter
          </a>
          <a
            href="https://www.youtube.com/mixeria-milionaria"
            target="_blank"
            rel="noopenner, noreferrer"
          >
            Youtube
          </a>
          <a
            href="https://www.facebook.com/mixeria-milionaria"
            target="_blank"
            rel="noopenner, noreferrer"
          >
            Facebook
          </a>
        </nav>
      </S.Column>

      <S.Column aria-labelledby="footer-resources">
        <S.Heading color="black">Links</S.Heading>

        <nav>
          <Link href="/">
            <a>Inicio</a>
          </Link>
          <Link href="/mixarias">
            <a>Mixarias</a>
          </Link>
          <Link href="/buscar">
            <a>Buscar</a>
          </Link>
        </nav>
      </S.Column>

      <S.Column aria-labelledby="footer-contact">
        <S.Heading color="black">Location</S.Heading>
        <span>Lorem ipsum dolor sit.</span>
        <span>Lorem Ipsum</span>
        <span>Lorem, ipsum dolor.</span>
      </S.Column>
    </S.Content>

    <S.Copyright>Mixaria Milionaria 2021 Â© All rights reserved.</S.Copyright>
  </S.Wrapper>
)

export default Footer
