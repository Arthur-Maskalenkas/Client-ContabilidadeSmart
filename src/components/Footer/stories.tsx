import { Story, Meta } from '@storybook/react/types-6-0'
import Footer from '.'

export default {
  title: 'Footer',
  component: Footer,
  parameters: {
    backgrounds: {
      default: 'mm-light'
    }
  }
} as Meta

export const Default: Story = () => (
  <div style={{ maxWidth: '100rem', margin: '0 auto' }}>
    <Footer />
  </div>
)
