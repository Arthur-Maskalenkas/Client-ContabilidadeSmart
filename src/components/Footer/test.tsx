import { render, screen } from 'utils/test-utils'

import Footer from '.'

describe('<Footer />', () => {
  it('vai renderizar os icones', () => {
    render(<Footer />)

    expect(screen.getByRole('svg', { name: /Linkedin/i })).toBeInTheDocument()
    expect(screen.getByRole('svg', { name: /Instagram/i })).toBeInTheDocument()
    expect(screen.getByRole('svg', { name: /Whatsapp/i })).toBeInTheDocument()
    expect(screen.getByRole('svg', { name: /Gmail/i })).toBeInTheDocument()
  })
})
