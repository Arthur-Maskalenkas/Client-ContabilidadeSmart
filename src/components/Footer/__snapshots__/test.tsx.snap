// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render 4 column topics 1`] = `
.c4 {
  color: #050706;
  font-size: 1.6rem;
}

.c4::after {
  width: 3rem;
}

.c6 {
  color: #050706;
}

.c0 .c3 {
  text-transform: uppercase;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(2,1fr);
  gap: 3.2rem;
  margin-top: 3.2rem;
}

.c2 a,
.c2 span {
  display: block;
  color: #8F8F8F;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 0.8rem;
  font-size: 1.4rem;
}

.c2 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7 {
  color: #8F8F8F;
  font-size: 1.2rem;
  margin-top: 4.0rem;
  margin-bottom: 3.2rem;
  text-align: center;
}

.c5 {
  margin-bottom: 2.4rem;
}

@media (min-width:768px) {
  .c1 {
    grid-template-columns: repeat(4,1fr);
  }
}

<footer
  class="c0"
>
  <div
    class="c1"
  >
    <div
      aria-labelledby="column-contact"
      class="c2"
    >
      <h2
        class="c3 c4 c5"
        color="black"
      >
        Contact Us
      </h2>
      <a
        href="mailto:sac@wongames.com"
      >
        mm@mm.com
      </a>
    </div>
    <div
      aria-labelledby="social-media"
      class="c2"
    >
      <h2
        class="c3 c6 c5"
        color="black"
      >
        Follow us
      </h2>
      <nav>
        <a
          href="https://www.instagram.com/mixeria-milionaria"
          rel="noopenner, noreferrer"
          target="_blank"
        >
          Instagram
        </a>
        <a
          href="https://www.twitter.com/mixeria-milionaria"
          rel="noopenner, noreferrer"
          target="_blank"
        >
          Twitter
        </a>
        <a
          href="https://www.youtube.com/mixeria-milionaria"
          rel="noopenner, noreferrer"
          target="_blank"
        >
          Youtube
        </a>
        <a
          href="https://www.facebook.com/mixeria-milionaria"
          rel="noopenner, noreferrer"
          target="_blank"
        >
          Facebook
        </a>
      </nav>
    </div>
    <div
      aria-labelledby="footer-resources"
      class="c2"
    >
      <h2
        class="c3 c6 c5"
        color="black"
      >
        Links
      </h2>
      <nav>
        <a
          href="/"
        >
          Inicio
        </a>
        <a
          href="/mixarias"
        >
          Mixarias
        </a>
        <a
          href="/buscar"
        >
          Buscar
        </a>
      </nav>
    </div>
    <div
      aria-labelledby="footer-contact"
      class="c2"
    >
      <h2
        class="c3 c6 c5"
        color="black"
      >
        Location
      </h2>
      <span>
        Lorem ipsum dolor sit.
      </span>
      <span>
        Lorem Ipsum
      </span>
      <span>
        Lorem, ipsum dolor.
      </span>
    </div>
  </div>
  <div
    class="c7"
  >
    Mixaria Milionaria 2021 Â© All rights reserved.
  </div>
</footer>
`;
